ex03.exs

```diff
@@ -61,8 +62,23 @@ defmodule Ex03 do
 
   """
 
+  def map_chunks(chunks, function) do
+    Enum.map(chunks, fn chunk ->
+      Task.async(fn ->
+        Enum.map(chunk, &( function.(&1) ))
+      end)
+    end)
+  end
+
   def pmap(collection, process_count, function) do
-    « your code here »
+    chunk_count = div(Enum.count(collection), process_count)
+
```

Comment:
This is a fairly picky issue, but it's something I feel is important. If you look at this code:

~~~ elixir
    |> Enum.to_list()
    |> Enum.chunk_every(chunk_count)
    |> map_chunks(function)
    |> Enum.map(fn task -> Task.await(task) end)
    |> Enum.concat()
~~~

you can see that, to paraphrase Sesame Street, one of these things is not like the others.

In particular, look at

~~~ elixir
    |> map_chunks(function)
    |> Enum.map(fn task -> Task.await(task) end)
~~~

The first line delegates its work to a function (`map_chunks`) which hides the actual implementation. The second function then exposes the implementation to be task-based by using `await`.

This makes the code tricky to read. You need to understand two functions together before making sense of either.

Be careful to keep the level of abstraction the same throughout a function (or any piece of code)